src/shared/locale-setup.c

```diff
@@ -208,11 +212,9 @@ int locale_context_save(LocaleContext *c, char ***ret_set, char ***ret_unset) {
         if (r < 0)
                 return r;
 
-        if (stat("/etc/locale.conf", &st) < 0)
+        if (stat("/etc/locale.conf", &c->st) < 0)
```

Comment:
Hmm, but this is also a TOCTOU race. Maybe it's not worth fixing…

I think we could have `write_env_file_full()` that'd take a `struct stat *` and return that stat of the fd after `flush_and_check()`.

Reply:
Yeah, possibly. I will try to address that later in another PR.